modules/platform/nuxeo-template-rendering/nuxeo-template-rendering-core/src/test/java/org/nuxeo/template/TestTemplateRenderingService.java

```diff
@@ -274,4 +279,26 @@ protected TemplateSourceDocument createTemplateSourceDoc(String templateString,
 
         return templateSource.getAdapter(TemplateSourceDocument.class);
     }
+
+    @Test
+    public void testFormatDate() {
+        String default_format = "MM/dd/yyyy";
+        String format1 = "yyyy MMM dd HH:mm:ss";
+        String format2 = "EEEE dd MMMMM yyyy";
+        Locale locale = Locale.getDefault();
+
+        GregorianCalendar calendar = new GregorianCalendar(2013, 11, 31);
+        Date dt = calendar.getTime();
+
+        SimpleDateFormat default_sdf = new SimpleDateFormat(default_format, locale);
+        SimpleDateFormat sdf1 = new SimpleDateFormat(format1, locale);
+        SimpleDateFormat sdf2 = new SimpleDateFormat(format2, locale);
+
+        ContextFunctions cf = new ContextFunctions(null, null);
+
+        assertEquals(cf.formatDate(calendar), default_sdf.format(dt));
+        assertEquals(cf.formatDate(calendar, format1), sdf1.format(dt));
+        assertEquals(cf.formatDate(calendar, format2), sdf2.format(dt));
```

Comment:
When using `assertEquals` the first argument should be the "expected" value (here `sdf.format(dt)`) so that the error messages make sense in case of differences.